test/tests/pbs_Rrecord_resources_used.py

```diff
@@ -0,0 +1,616 @@
+        test1 += ['sleep 1000\n']
+
+        """
+        Submit three jobs J1,J2,J3[]
+        """
+        j1 = Job(TEST_USER, attrs={ATTR_k: 'oe'})
+        j1.create_script(body=test)
+        jid1 = self.server.submit(j1)
+
+        if is_nonrerunnable==True:
+            j2 = Job(TEST_USER, attrs={ATTR_r: 'n', ATTR_k: 'oe' })
+        else:
+            j2 = Job(TEST_USER, attrs={ATTR_r: 'y', ATTR_k: 'oe'})
+
+        j2.create_script(body=test1)
+        jid2 = self.server.submit(j2)
+
+        j3 = Job(TEST_USER, attrs={ATTR_J: '1-6:1', ATTR_k: 'oe'})
+        j3.create_script(body=test)
+        jid3 = self.server.submit(j3)
+
+        subjobs = self.server.status(JOB, id=jid3, extend='t')
+        jid3s1 = subjobs[1]['id']
+
+        """
+        Wait for the jobs to start running.
+        """
+        self.server.expect(JOB, {ATTR_state: 'R'}, jid1, offset=15)
+        self.server.expect(JOB, {ATTR_state: 'R'}, jid2, offset=15)
+        self.server.expect(JOB, {ATTR_state: 'R'}, jid3s1, offset=15)
+
+        """
+        Verify that accounting logs have Resource_List.<resource> value
+        """
+        m = self.server.accounting_match(
+        msg='.*Resource_List.*', id=jid1, regexp=True)
+        self.assertNotEqual(m, None)
+        m = self.server.accounting_match(
+        msg='.*Resource_List.*', id=jid2, regexp=True)
+        self.assertNotEqual(m, None)
+        m = self.server.accounting_match(
+        msg='.*Resource_List.*', id=jid3s1, regexp=True)
+        self.assertNotEqual(m, None)
+
+        """
+        Bring both moms down using kill -9 < mom pid>
+        """
+        self.momA.signal('-KILL')
+        self.momB.signal('-KILL')
+
+        """
+        Verify that both nodes are reported to be down.
+        """
+        self.server.expect(NODE, {ATTR_NODE_state: (
+        MATCH_RE, '.*down.*')}, id=self.hostA)
+        self.server.expect(NODE, {ATTR_NODE_state: (
+        MATCH_RE, '.*down.*')}, id=self.hostB)
+
+        if is_nonrerunnable==False:
+            """
+            All rerunnable jobs - all should be in 'Q' state.
+            """
+            self.server.expect(JOB, {ATTR_state: 'Q'}, jid1, offset=5)
+            self.server.expect(JOB, {ATTR_state: 'Q'}, jid2)
+            self.server.expect(JOB, {ATTR_state: 'Q'}, jid3s1)
+        else:
+            """
+            Job2 is non-rerunnable.
+            """
+            self.server.expect(JOB, {ATTR_state: 'Q'}, jid1, offset=5)
+            self.server.expect(JOB, {ATTR_state: 'F'}, jid2, extend='x')
+            self.server.expect(JOB, {ATTR_state: 'Q'}, jid3s1)
+
+        """
+        tracejob should show "Job requeued, execution node <node name> down"
+        """
+        m = self.server.tracejob_match(
+        msg='Job requeued, execution node .* down', id=jid1, regexp=True)
+        self.assertNotEqual(m, None)
+
+        if is_nonrerunnable==False:
+            m = self.server.tracejob_match(
```

Comment:
Maybe move the log match outside of the if and only have the if check the assert.